src/game/level.c

```diff
@@ -925,6 +928,39 @@ static void Level_CompleteSetup(int32_t level_num)
     Memory_FreePointer(&sample_sizes);
 }
 
+static void Level_CalculateMaxVertices(void)
+{
+    size_t max_vertices = 0;
+    for (int32_t i = 0; i < O_NUMBER_OF; i++) {
+        const OBJECT_INFO *object_info = &g_Objects[i];
+        if (!object_info->loaded) {
+            continue;
+        }
+
+        for (int32_t j = 0; j < object_info->nmeshes; j++) {
+            max_vertices =
+                MAX(max_vertices, *(g_Meshes[object_info->mesh_index + j] + 5));
+        }
+    }
+
+    for (int32_t i = 0; i < STATIC_NUMBER_OF; i++) {
+        const STATIC_INFO *static_info = &g_StaticObjects[i];
+        if (!static_info->loaded || static_info->nmeshes < 0) {
+            continue;
+        }
+
+        max_vertices =
+            MAX(max_vertices, *(g_Meshes[static_info->mesh_number] + 5));
+    }
+
+    for (int32_t i = 0; i < g_RoomCount; i++) {
+        max_vertices = MAX(max_vertices, *g_RoomInfo[i].data);
+    }
+
+    LOG_INFO("Maximum vertices: %d", max_vertices);
+    Output_ReserveVertexBuffer(max_vertices);
+}
```

Comment:
Let's have this function return an `int32_t` and call the `Output_…` function inside higher up in the hierarchy.